### PR TITLE
#161404774 Add all product related endpoints

### PR DESCRIPTION
**Pull request by kwanj-k**
### What does this PR do?
Adds all the product related endpoints

#### Description of Task to be Completed?
 * The PR adds the following endpoints

| EndPoint                                                   | Functionality                                       |
| ------------------------------------------------------|:-------------------------------------------------:|                                                                 
| POST     /products/                                   | Add a product                                     | 
|GET       /products/                                    | Get all products                                   | 
| PUT      /products/Id/                                | Update the information of a product    |
| POST      /products/Id/                             |    Add a product to cart                         |
| DELETE   /products/Id/                            | Remove a product                                |
| GET      /cart/                                            | Get my cart                                           |
| POST      /cart/                                          | Sell an entire cart                                 |
| PUT      /cart/Id/                                         | Update a product in a cart                    |
| DELETE   /cart/id                                      | Delete a product from a cart                  |
| DELETE   /cart/                                         | Delete an entire cart                               |
| GET      /sales/                                          | Get all sales                                            |
| POST     /categories/                                |  Create a category                                   | 
|GET       /categories/                                 | Get all categories                                     | 
| PUT      /categories/Id/                             | Update a category                                    |
| DELETE   /categories/Id/                         | Remove a category                                   |
| POST    /categories/Id//products/Id/        |  Give a product a given category               |

### How should this be manually tested?
* After cloning the repo, git pull the branch ft-products-#161404774
* Install a virtual enviroment and activate it [Refer here](https://docs.python-guide.org/dev/virtualenvs/)
* Pip install the the requirements.txt
* Run the following:
  ``` export FLASK_APP='run.py' ```
  ``` export APP_SETTINGS='development' ```
   ``` pytest --cov-report term-missing --cov=app ```

Also, you need to create a development and test database and connect to it.[Refer here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-18-04#creating-a-new-role)

Then export the database URL in the environment:
```
export Dev_URL="dbname=dbname user=username password=password"
export Test_URL="dbname=dbname user=username password=password"
export SECRET="iamasecrettheunsadtheunsulliedtheundead"

```
We need the secret key for generating the tokens

Then Run the tests:
```
pytest --cov-report term-missing --cov=app
```
* All the tests should pass:
screenshot from 2018-10-25 20-23-04

### What are the relevant Pivotal Tracker stories?
#161404774

### CODE REVIEW

File: app/api/common/validators.py [hunk 1]
```diff
@@ -86,4 +140,3 @@ def decorator(*args, **kwargs):
             abort(406, msg)
         return f(*args, **kwargs)
     return decorator
-
```
Comment (dotNesh) [inline app/api/common/validators.py]: Leave a line at the bottom of the file.

File: app/api/v2/db_config.py [hunk 2]
```diff
@@ -30,39 +31,76 @@ def create_tables():
...
+    categories = """
+        CREATE TABLE IF NOT EXISTS categories(id serial PRIMARY KEY,
+        store_id int,
+        name varchar,
+        created_at varchar);
+        """
+
+    queries = [stores, users, products, sales, carts, categories]
     return queries
-
```
Comment (dotNesh) [inline app/api/v2/db_config.py:68]: Leave an empty line at the bottom of the file.

File: app/api/v2/views/auth.py [hunk 3]
```diff
@@ -4,65 +4,67 @@
...
         Create a store
         """
-        json_data = request.get_json(force = True)
+        json_data = request.get_json(force=True)
         new_store_validator(json_data)
         storecheck = get_store_by_name(json_data['name'])
         if storecheck:
             msg = 'Store name already exists'
-            abort(406,msg)
+            abort(406, msg)
```
Comment (dotNesh) [inline app/api/v2/views/auth.py]: I don't think the abort is necessary, just return the response message.

Comment (dotNesh) [inline app/api/v2/views/auth.py]: The status code for an existing thing is 409.

**Review by dotNesh — COMMENTED** [review]
Work on the requested changes.

File: app/api/v2/db_config.py (hunk 2 above)
Comment (kwanj-k) [reply app/api/v2/db_config.py:68]: This file has refused a new line
But I will keep trying

**Pull request merged**